src/compiler/commandLineParser.ts

```diff
@@ -509,12 +509,14 @@ namespace ts {
     let optionNameMapCache: OptionNameMap;
 
     /* @internal */
-    export function replaceEnableAutoDiscoveryWithEnable(typeAcquisition: TypeAcquisition): void {
-        // Replace deprecated typingOptions.enableAutoDiscovery with typeAcquisition.enable
+    export function convertEnableAutoDiscoveryToEnable(typeAcquisition: TypeAcquisition): TypeAcquisition {
+        // Convert deprecated typingOptions.enableAutoDiscovery to typeAcquisition.enable
+        const result = typeAcquisition;

```

Comment:
this is effectively an alias, so any modifications on `result` will affect source `typeAcquisition` object